src/keyring/KeyringController.ts

```diff
@@ -347,17 +352,18 @@ export class KeyringController extends BaseController<KeyringConfig, KeyringStat
    */
   async signTypedMessage(messageParams: TypedMessageParams, version: string) {
     try {
-      const address = sigUtil.normalize(messageParams.from);
+      const address = normalizeAddress(messageParams.from);
       const { password } = privates.get(this).keyring;
       const privateKey = await this.exportAccount(password, address);
       const privateKeyBuffer = ethUtil.toBuffer(ethUtil.addHexPrefix(privateKey));
       switch (version) {
         case 'V1':
-          return sigUtil.signTypedDataLegacy(privateKeyBuffer, { data: messageParams.data });
+          // signTypedDataLegacy will throw if the data is invalid.
```

Comment:
I'm curious about why this was added - is this a change in behavior? Or is the fact that this can throw unexpected for some reason?

Reply:
This is in a `try` block. The comment was added to explain the cast to `any`.

Reply:
Ahhh I see - I missed the cast